src/package-linker.js

```diff
@@ -251,10 +251,41 @@ export default class PackageLinker {
       }
     }
 
-    // linked modules
-    for (const loc of possibleExtraneous) {
-      const stat = await fs.lstat(loc);
+    // if an extraneous is an entry created via "yarn link", we prevent it from being overwritten
+    // unfortunately, the only mean we have to know if they have been created this way is to check if
+    // they are a symlink - problem is that it then conflicts with the newly introduced "link:" protocol,
+    // which also creates symlinks :( a somewhat weak fix is to check if the symlink target is registered
+    // inside the linkFolder, in which case we assume it has been created via "yarn link". Otherwise, we
+    // remove it as usual.
+    const linkTargets = new Map();
+
+    for (const entry of await fs.readdir(this.config.linkFolder)) {
+      const entryPath = path.join(this.config.linkFolder, entry);
+      const stat = await fs.lstat(entryPath);
       if (stat.isSymbolicLink()) {
+        linkTargets.set(entry, await fs.readlink(entryPath));
+      } else if (stat.isDirectory()) {
+        for (const entry2 of await fs.readdir(entryPath)) {
+          const entryPath2 = path.join(entryPath, entry2);
+          const stat2 = await fs.lstat(entryPath2);
+          if (stat2.isSymbolicLink()) {
+            linkTargets.set(`${entry}/${entry2}`, await fs.readlink(entryPath2));
+          }
+        }
+      }
+    }
+
+    for (const loc of possibleExtraneous) {
+      const basename = path.basename(loc);
+      const parentname = path.basename(path.dirname(loc));
```

Comment:
I'm not sure it would be much clearer -

```js
const parsedPath = path.parse(loc);

const basename = parsedPath.base;
const parentname = path.basename(parsedPath.dir);
```